### PR TITLE
Ask user for confirmation when deleting multiple uses of a variable.

### PR DESCRIPTION
**Pull request by rachel-fenichel**

### CODE REVIEW

File: core/variables.js [hunk 1]
```diff
@@ -267,11 +268,9 @@ Blockly.Variables.getUses = function(name, workspace) {
 
 /**
  * When a variable is deleted, find and dispose of all uses of it.
- * @param {string} name Name of deleted variable.
- * @param {!Blockly.Workspace} workspace The workspace to delete uses from.
+ * @param {!Array.<!Blockly.Block>} uses An array of blocks using the variable.
  */
-Blockly.Variables.disposeUses = function(name, workspace) {
-  var uses = Blockly.Variables.getUses(name, workspace);
+Blockly.Variables.disposeUses = function(uses) {
```
Comment (picklesrus) [inline core/variables.js]: was this public?  If so, we should be careful of changing the API and maybe deprecate the old version in favor of the new one?  I'm not sure what the general practice is for deprecating stuff in blockly... but I have seen the deprecated tag :)

File: core/variables.js [hunk 2]
```diff
@@ -287,10 +286,16 @@ Blockly.Variables.disposeUses = function(name, workspace) {
 Blockly.Variables.delete = function(name, workspace) {
   var variableIndex = workspace.variableList.indexOf(name);
   if (variableIndex != -1) {
+    var uses = Blockly.Variables.getUses(name, workspace);
+    if (uses.length > 1) {
+      window.confirm(
+        Blockly.Msg.DELETE_VARIABLE_CONFIRMATION.replace('%1', uses.length).
+            replace('%2', name));
+    }
+    Blockly.Variables.disposeUses(uses, workspace);
```
Comment (picklesrus) [inline core/variables.js]: should you remove the workspace param here?

Comment (RoboErikG) [inline core/variables.js]: workspace isn't a parameter anymore

File: core/variables.js (hunk 1 above)
Comment (RoboErikG) [inline core/variables.js]: I think the old API was a bit better for working with. It let developers call it without figuring out how to get the full list of blocks using it. Maybe make it just a lightweight wrapper that calls a private version that takes the list of uses so you can do the optimization below?

File: core/variables.js (hunk 2 above)
Comment (rachel-fenichel) [inline core/variables.js]: Thanks, javascript!

File: core/variables.js (hunk 1 above)
Comment (rachel-fenichel) [inline core/variables.js]: This was public in this branch, but I think it should be private.  It doesn't exist at all in code that anyone else is using--remember that variable deletion is not a concept in current blockly, so of course there's no API for it.

Erik, I'm confused by your comment on the old API.  I think what you want is just a call to Variables.delete, which does exactly what you're saying and is public.

Comment (picklesrus) [inline core/variables.js]: Sorry, I didn't look closely enough at the PR to notice you were on the variable branch.  Do you want to mark it private now?

**Comment by picklesrus** [comment]
:lgtm:

---

Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/526#-:-KOlfFmGXP7Q958QI0n1:bnfp4nl)*

**Pull request merged**